Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Motion/MotionDataBuilder.cpp

```diff
@@ -347,11 +347,12 @@ namespace EMotionFX
                         // For additive motion, we stores the relative transform.
                         boneTransform = sampleFrameTransformInverse * boneTransform;
                     }
-                    
+
                     SceneAPIMatrixType boneTransformNoScale(boneTransform);
-                    const AZ::Vector3 position = coordSysConverter.ConvertVector3(boneTransform.GetTranslation());
-                    const AZ::Quaternion rotation = coordSysConverter.ConvertQuaternion(AZ::Quaternion::CreateFromMatrix3x4(boneTransformNoScale));
-                    const AZ::Vector3 scale = coordSysConverter.ConvertScale(boneTransformNoScale.ExtractScale());
+                    const AZ::Vector3 scale = boneTransformNoScale.ExtractScale();
+                    const AZ::Transform convertedTransform = AZ::Transform::CreateFromMatrix3x4(coordSysConverter.ConvertMatrix3x4(boneTransformNoScale));
+                    const AZ::Vector3 position = convertedTransform.GetTranslation();
+                    const AZ::Quaternion rotation = convertedTransform.GetRotation();
```

Comment:
suggestion: I realize this is a bit of a monster function, so it may be difficult to test, but actually testing just this small piece in isolation (if we make a free function that takes this data as parameters) we can then easily recreate the 'bad' data/case before, and prove the updated version behaves correctly, test that, and then call the function here. This is a reasonable way to verify this functionality when writing a larger integration test may be more complex (testing free functions with just inputs/outputs is easy 🙂). I'd highly recommend this refactor and test before merging if possible